crates/primitives/src/block.rs

```diff
@@ -222,49 +223,30 @@ impl<'de> Deserialize<'de> for BlockId {
                 A: MapAccess<'de>,
             {
                 let mut number = None;
-                let mut hash = None;
-                let mut hash_canonical = None;
+                let mut block_hash = None;
+                let mut require_canonical = None;
 
                 while let Some(key) = map.next_key::<String>()? {
                     match key.as_str() {
                         "blockNumber" => {
-                            if number.is_some() || hash.is_some() {
+                            if number.is_some() || block_hash.is_some() {
```

Comment:
require_canonical must be None as well.

Reply:
Why whould I have to check for require_canonical to be None here? 🤔

Reply:
`require_canonical` is not allowed in the `blockNumber` context